### PR TITLE
Fix compile with option TIFLASH_ENABLE_AVX_SUPPORT=OFF

### LINKED ISSUE #6877: Compile error when building with option `TIFLASH_ENABLE_AVX_SUPPORT=OFF`
## Enhancement

Compile error when compiling without `AVX2` feature support using cmake option `-DTIFLASH_ENABLE_AVX_SUPPORT=OFF`

```
In file included from /home/dev/workspace/tiflash-upstream/libs/libcommon/src/StringRef.cpp:17:
In file included from /home/dev/workspace/tiflash-upstream/libs/libcommon/include/common/StringRef.h:19:
/home/dev/workspace/tiflash-upstream/libs/libcommon/include/common/mem_utils_opt.h:87:17: error: no member named 'memcmp' in namespace 'std'; did you mean simply 'memcmp'?
    return 0 == std::memcmp(lhs.data(), rhs.data(), lhs.size());
                ^~~~~~~~~~~
                memcmp
/usr/include/string.h:64:12: note: 'memcmp' declared here
extern int memcmp (const void *__s1, const void *__s2, size_t __n)
           ^
1 error generated.
[3576/4521] Building CXX object libs/libcommon/CMakeFiles/common.dir/src/DateLUT.cpp.o
ninja: build stopped: subcommand failed.
```

### PR DESCRIPTION
**Pull request by yongman**
### What problem does this PR solve?

Issue Number: close #6877 

Problem Summary:

### What is changed and how it works?

Add `TIFLASH_ENABLE_AVX_SUPPORT` check in `ToBits64` and `CountBytesInFilter`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Common/StringUtils/StringUtils.h [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #include <cstring>
 #include <string>
 #include <string_view>
+#include <tuple>
```
Comment (Lloyd-Pottiger) [inline dbms/src/Common/StringUtils/StringUtils.h:22]: why add this?

**Review by Lloyd-Pottiger — APPROVED** [review]

**Comment by solotzg** [comment]
You can cmake option `NO_SSE42_OR_HIGHER` to disable features avx, avx2, ... and later.

**Comment by solotzg** [comment]
@Lloyd-Pottiger `NO_SSE42_OR_HIGHER` it looks like a typo. We still need sse4.2 by default.

**Comment by Lloyd-Pottiger** [comment]
> You can cmake option `NO_SSE42_OR_HIGHER` to disable features avx, avx2, ... and later.

when we set flag `-mavx2` to a file, the `__AVX2__  = 1` whether the machine support `avx2` or not. We may still need `TIFLASH_ENABLE_AVX_SUPPORT` to dispatch.

> @Lloyd-Pottiger `NO_SSE42_OR_HIGHER` it looks like a typo. We still need sse4.2 under AMD64 platforms by default.

yes, it is a typo. @yongman could you please help rename `NO_SSE42_OR_HIGHER` to `NO_AVX_OR_HIGHER` in this pr?

**Comment by solotzg** [comment]
-march=haswell or later platform  can also define macro __AVX2__. Since avx2 is the default required feature of tiflash, I think macro __AVX2__ is enough.

**Comment by solotzg** [comment]
We need to guarantee that if the platform used for compiling can not support avx2, NO_SSE42_OR_HIGHER should be set to ON.

**Comment by yongman** [comment]
@solotzg  Set `NO_SSE42_OR_HIGHER` to `ON` also has some errors for now, I will check if it works after another fix PR.

**Comment by Lloyd-Pottiger** [comment]
> We need to guarantee that if the platform used for compiling can not support avx2, NO_SSE42_OR_HIGHER should be set to ON.

[check_cxx_source_runs](https://cmake.org/cmake/help/latest/module/CheckCXXSourceRuns.html) may help, but it only exists in v3.26.0 or higher of cmake.

**Comment by solotzg** [comment]
> @solotzg Set `NO_SSE42_OR_HIGHER` to `ON` also has some errors for now, I will check if it works after another fix PR.

To be fixed by #6879

**Pull request closed without merging**